src/clients/QBittorrent.ts

```diff
@@ -329,38 +327,55 @@ export default class QBittorrent implements TorrentClient {
 				(await this.isSubfolderContentLayout(searchee))
 					? "Subfolder"
 					: "Original";
-
-			const skipRecheckTorrent =
-				decision === Decision.MATCH_PARTIAL ? skipRecheck : true;
-
 			const formData = new FormData();
 			formData.append("torrents", buffer, filename);
-			formData.append("tags", TORRENT_TAG);
-			formData.append("category", newCategoryName);
-
-			if (autoTMM) {
-				formData.append("autoTMM", "true");
-			} else {
-				formData.append("autoTMM", "false");
-				formData.append("savepath", save_path);
-			}
+			// searchee was linked
 			if (path) {
-				formData.append("contentLayout", "Original");
-				formData.append("skip_checking", skipRecheck.toString());
-				formData.append("paused", (!skipRecheck).toString());
+				//because it was a linked file, turn off autotmm
+				//and set a save path and datacat
+				formData.append(
+					"autoTMM",
+					flatLinking && searchee.infoHash ? autoTMM : "false",
+				);
+				formData.append("savepath", save_path);
+				formData.append("category", linkCategory);
+				// tag category
+				formData.append(
+					"tags",
+					searchee.infoHash
+						? `${TORRENT_TAG},${newCategoryName}`
+						: category,
+				);
 			} else {
-				formData.append("contentLayout", contentLayout);
-				formData.append("skip_checking", skipRecheckTorrent.toString());
-				formData.append("paused", (!skipRecheckTorrent).toString());
+				// torrent-backed
+				// use searchee's autoTMM
+				// we infer this based on no path (linking) being done
+				formData.append("autoTMM", autoTMM.toString());
+				//set category and tags like v5 perfectMatch
+				formData.append("category", newCategoryName);
+				formData.append(
+					"tags",
+					searchee.infoHash
+						? TORRENT_TAG
+						: `${TORRENT_TAG},${newCategoryName}`,
+				);
 			}
-
+			formData.append("contentLayout", path ? "Original" : contentLayout);
+			formData.append(
+				"skip_checking",
+				determineSkipRecheck(decision).toString(),
+			);
+			formData.append(
+				"paused",
+				!determineSkipRecheck(decision).toString(),
+			);
 			// for some reason the parser parses the last kv pair incorrectly
 			// it concats the value and the sentinel
 			formData.append("foo", "bar");
 
 			await this.request("/torrents/add", formData);
-
-			if (path && !skipRecheck) {
+			//if we have a linked file and skiprecheck is false
+			if (determineSkipRecheck(decision)) {
```

Comment:
why are we no longer checking for path here?

Reply:
my thought was that because path was previously exclusive to data-based matching, which we always honored skipRecheck with, but was now solely dependent on whether or not linking was enabled and configured properly to link the searchee, we would just depend on the logic in `determineSkipRecheck`.

Even if it is torrent-backed, we would still be provided a `path` in some (or most if configured) . There is no reason, in my opinion, to calculate rechecking conditional on linking, since it could be a linked torrent-backed searchee we would previously have skipped rechecking on in v5.

I altered the logic to depend solely on the `matchMode`, except for `risky` matches, in which case it would honor `skipRecheck`.

Thoughts?

Reply:
good change, and well reasoned.